hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/AbstractClientSelectionHandler.java

```
@@ -51,12 +51,8 @@ public final void onFailure(Throwable e) {
             sk.cancel();
         }
         connectionManager.destroyConnection(connection);
-        StringBuilder sb = new StringBuilder();
-        sb.append(Thread.currentThread().getName());
-        sb.append(" Closing socket to endpoint ");
-        sb.append(connection.getEndPoint());
-        sb.append(", Cause:").append(e);
-        logger.warning(sb.toString());
+        logger.warning(Thread.currentThread().getName() + " Closing socket to endpoint "
+                + connection.getEndPoint() + ", Cause:" + e);
```

Perhaps better; but not part of this PR. I also hate swallowing exceptions in such a way.